app/src/main/java/org/p2p/wallet/auth/username/repository/UsernameParser.kt

```diff
@@ -0,0 +1,27 @@
+package org.p2p.wallet.auth.username.repository
+
+import org.p2p.wallet.auth.model.Username
+import org.p2p.wallet.auth.username.repository.model.UsernameDetails
+import org.p2p.wallet.common.feature_toggles.toggles.remote.UsernameDomainFeatureToggle
+import org.p2p.wallet.utils.Base58String
+
+/**
+ * @see [org.p2p.wallet.auth.username.repository.UsernameParserTest]
+ */
+class UsernameParser(
+    private val usernameDomainFeatureToggle: UsernameDomainFeatureToggle
+) {
+    /**
+     * @param anyUsername - username with domain prefix or not
```

Comment:
Did you ask if we really need this logic at all? 
I don't think feature toggle is needed, moreover the domain like `.key.sol`. 

Our business couldn't own this domain as planned, so looks like the feature toggle and this logic is redundant at all